### PR TITLE
PLT-7145: `platform user deleteall` expects a user as argument

### PR DESCRIPTION
**Pull request by MusikPolice**
Please make sure you've read the [pull request](http://docs.mattermost.com/developer/contribution-guide.html#preparing-a-pull-request) section of our [code contribution guidelines](http://docs.mattermost.com/developer/contribution-guide.html).

When filling in a section please remove the help text and the above text.

#### Summary
Fixed typo preventing deleteall command from working as expected. Cleaned up spelling and error messages throughout the user module

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-7145
https://github.com/mattermost/platform/issues/6925

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

### CODE REVIEW

File: cmd/platform/user.go [hunk 1]
```diff
@@ -86,7 +85,7 @@ var deleteAllUsersCmd = &cobra.Command{
 	Short:   "Delete all users and all posts",
 	Long:    "Permanently delete all users and all related information including posts.",
 	Example: "  user deleteall",
-	RunE:    deleteUserCmdF,
+	RunE:    deleteAllUsersCommandF,
```
Comment (MusikPolice) [inline cmd/platform/user.go]: This was the actual bug. Everything else in the file is just cleanup

File: cmd/platform/user.go [hunk 2]
```diff
@@ -187,7 +186,7 @@ func changeUserActiveStatus(user *model.User, userArg string, activate bool) err
 		return fmt.Errorf("Can't find user '%v'", userArg)
 	}
 	if user.IsLDAPUser() {
-		return errors.New(utils.T("api.user.update_active.no_deactivate_ldap.app_error"))
+		return errors.New("You can not modify the activation status of AD/LDAP accounts. Please modify through the AD/LDAP server.")
```
Comment (MusikPolice) [inline cmd/platform/user.go]: This was the only string in the file that was being localized. Just grabbed the string from the en file to replace it

File: cmd/platform/user.go [hunk 3]
```diff
@@ -230,7 +229,7 @@ func userCreateCmdF(cmd *cobra.Command, args []string) error {
 	firstname, _ := cmd.Flags().GetString("firstname")
 	lastname, _ := cmd.Flags().GetString("lastname")
 	locale, _ := cmd.Flags().GetString("locale")
-	system_admin, _ := cmd.Flags().GetBool("system_admin")
+	systemAdmin, _ := cmd.Flags().GetBool("system_admin")
```
Comment (MusikPolice) [inline cmd/platform/user.go]: Go variable names aren't supposed to have underscores in them

**Comment by crspeller** [comment]
Setting to PM review for text changes.

File: cmd/platform/user.go [hunk 4]
```diff
@@ -128,14 +127,14 @@ var searchUserCmd = &cobra.Command{
...
-	userCreateCmd.Flags().String("nickname", "", "Nickname")
-	userCreateCmd.Flags().String("firstname", "", "First Name")
-	userCreateCmd.Flags().String("lastname", "", "Last Name")
-	userCreateCmd.Flags().String("locale", "", "Locale (ex: en, fr)")
-	userCreateCmd.Flags().Bool("system_admin", false, "Make the user a system administrator")
+	userCreateCmd.Flags().String("username", "", "Required. Username for the new user account")
+	userCreateCmd.Flags().String("email", "", "Required. The email address for the new user account")
+	userCreateCmd.Flags().String("password", "", "Required. The password for the new user account")
+	userCreateCmd.Flags().String("nickname", "", "Optional. The nickname for the new user account")
+	userCreateCmd.Flags().String("firstname", "", "Optional. The given name for the new user account")
```
Comment (jasonblais) [inline cmd/platform/user.go]: Propose using "first name" to follow Mattermost account settings

File: cmd/platform/user.go [hunk 5]
```diff
@@ -128,14 +127,14 @@ var searchUserCmd = &cobra.Command{
...
+	userCreateCmd.Flags().String("lastname", "", "Optional. The family or hereditary name for the new user account")
```
Comment (jasonblais) [inline cmd/platform/user.go]: Propose using "last name" to follow Mattermost account settings

File: cmd/platform/user.go [hunk 6]
```diff
@@ -128,14 +127,14 @@ var searchUserCmd = &cobra.Command{
...
 func init() {
-	userCreateCmd.Flags().String("username", "", "Username")
-	userCreateCmd.Flags().String("email", "", "Email")
-	userCreateCmd.Flags().String("password", "", "Password")
-	userCreateCmd.Flags().String("nickname", "", "Nickname")
-	userCreateCmd.Flags().String("firstname", "", "First Name")
-	userCreateCmd.Flags().String("lastname", "", "Last Name")
-	userCreateCmd.Flags().String("locale", "", "Locale (ex: en, fr)")
-	userCreateCmd.Flags().Bool("system_admin", false, "Make the user a system administrator")
+	userCreateCmd.Flags().String("username", "", "Required. Username for the new user account")
```
Comment (jasonblais) [inline cmd/platform/user.go]: Should these end in a period? 0/5

File: cmd/platform/user.go [hunk 7]
```diff
@@ -202,7 +201,7 @@ func userDeactivateCmdF(cmd *cobra.Command, args []string) error {
 	}
 
 	if len(args) < 1 {
-		return errors.New("Enter user(s) to deactivate.")
+		return errors.New("Expected at least one argument. See help text for details.")
```
Comment (jasonblais) [inline cmd/platform/user.go]: Which help text? Wondering if we can include which argument is expected, i.e. a user.

Similarly for the others

File: cmd/platform/user.go [hunk 8]
```diff
@@ -400,7 +399,7 @@ func deleteAllUsersCommandF(cmd *cobra.Command, args []string) error {
 		if confirm != "YES" {
 			return errors.New("ABORTED: You did not answer YES exactly, in all capitals.")
 		}
-		CommandPrettyPrintln("Are you sure you want to delete the teams specified?  All data will be permanently deleted? (YES/NO): ")
+		CommandPrettyPrintln("Are you sure you want to permanently delete all user accounts? (YES/NO): ")
```
Comment (jasonblais) [inline cmd/platform/user.go]: Thanks for cleaning these up!

File: cmd/platform/user.go (hunk 6 above)
Comment (MusikPolice) [inline cmd/platform/user.go]: I think it's a personal preference sort of thing, unless we have a standard for text like this. When printed out, they look like a list, and I think lists look kind of strange with punctuation like that.

File: cmd/platform/user.go (hunk 7 above)
Comment (MusikPolice) [inline cmd/platform/user.go]: Some of the messages were out of date. I think the arguments had changed since they were originally written. Others weren't clear on exactly what was expected. I opted to change the messages to reference the help doc so that we can avoid duplicating documentation in two places that might drift out of date from one another. I'm open to switching back if we think this is a problem

File: cmd/platform/user.go (hunk 6 above)
Comment (MusikPolice) [inline cmd/platform/user.go]: Fixed

File: cmd/platform/user.go (hunk 4 above)
Comment (MusikPolice) [inline cmd/platform/user.go]: Fixed

File: cmd/platform/user.go (hunk 5 above)
Comment (MusikPolice) [inline cmd/platform/user.go]: Fixed

**Comment by MusikPolice** [comment]
@jasonblais fixed a few issues, responded to others. Please take a look.

File: cmd/platform/user.go (hunk 7 above)
Comment (jasonblais) [inline cmd/platform/user.go]: Is it common to include a link to help documentation? (I'm not that familiar with the CLI tools).

If so, we could also link to it with, say, `Expected at least one argument. See documentation for details. [link]`

^I'm 0/5, we should do what's standard for CLI tools.

Comment (MusikPolice) [inline cmd/platform/user.go]: The help text that I'm referring to is what is printed if the user specifies the --h flag. There might be online documentation, but if there is, I'm not familiar with it. Users that are familiar with command line tools typically expect a man page or a --h flag to pull up documentation.

Comment (jasonblais) [inline cmd/platform/user.go]: Got it. That makes sense. As long as we're following the standard, which sounds like we do.

File: cmd/platform/user.go [hunk 9]
```diff
@@ -37,12 +37,11 @@ var userDeactivateCmd = &cobra.Command{
...
-	Use:   "create",
-	Short: "Create a user",
-	Long:  "Create a user",
-	Example: `  user create --email user@example.com --username userexample --password Password1
-  user create --firstname Joe --system_admin --email joe@example.com --username joe --password Password1`,
-	RunE: userCreateCmdF,
+	Use:     "create",
+	Short:   "Create a user",
+	Long:    "Create a user",
+	Example: `  user create --email user@example.com --username userexample --password Password1`,
```
Comment (crspeller) [inline cmd/platform/user.go]: Why did you remove the second example?

File: cmd/platform/user.go [hunk 10]
```diff
@@ -164,7 +163,7 @@ func userActivateCmdF(cmd *cobra.Command, args []string) error {
 	}
 
 	if len(args) < 1 {
-		return errors.New("Enter user(s) to activate.")
+		return errors.New("Expected at least one argument. See help text for details.")
```
Comment (crspeller) [inline cmd/platform/user.go]: Why change to leave out details?

File: cmd/platform/user.go (hunk 9 above)
Comment (MusikPolice) [inline cmd/platform/user.go]: It was unnecessary. The full set of options are documented in the --h text. The remaining example shows the minimum set of options required for the command to work. Anything else is optional and is marked as such.

File: cmd/platform/user.go (hunk 10 above)
Comment (MusikPolice) [inline cmd/platform/user.go]: Discussed above with @jasonblais. Some of the documentation was out of date or didn't match the --help text. My goal was to make a single source of information that users can rely on for the correct way to interpret errors.

**Comment by MusikPolice** [comment]
@crspeller Is there any remaining work that you'd like to see completed before I merge this?
